### PR TITLE
refactor(tester): Convert textlint-tester to TypeScript

### LINKED ISSUE #396: textlint-tester: Convert to TypeScript
[textlint-tester](https://github.com/textlint/textlint/tree/master/packages/textlint-tester "textlint-tester") is a test module for textlint rule.
It aim to test a textlint rule by declarative.

For that reason, this module has complex code. (has many optional option...)
We want to improve maintenancebility of the module.

First, We start to convert this module to TypeScript.

## Todo

- [ ] Convert src/ to TypeScript
  
Related:

- textlint: https://github.com/textlint/textlint/pull/332
- kernek: https://github.com/textlint/textlint/pull/301
- use ts-node: https://github.com/textlint/textlint/pull/354

### PR DESCRIPTION
**Pull request by azu**
fix #396

### CODE REVIEW

File: packages/textlint-tester/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import { TextLintTester } from "./textlint-tester";
+
+export = TextLintTester;
```
Comment (azu) [inline packages/textlint-tester/src/index.ts]: Remain `module.exports` for compatibility

File: packages/textlint-tester/test/ts-setup.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+require("ts-node").register({
+    project: __dirname
+});
```
Comment (0x6b) [inline packages/textlint-tester/test/ts-setup.js]: Do we need `typeCheck: true` here? (issue #448 and pull request #449, also ref. #451)

Comment (azu) [reply packages/textlint-tester/test/ts-setup.js]: Yes. Fix it

Comment (0x6b) [reply packages/textlint-tester/test/ts-setup.js]: Thanks!

**Comment by 0x6b** [comment]
LGTM

**Comment by azu** [comment]
Thanks

**Pull request merged**